### ISSUE TITLE
insert into 中包含 ; 时会插入报错

### ISSUE THREAD

**Issue by duopa** [StarRocks/starrocks#16526]
版本：2.3.7  
全部数据一起写入 报错：但是在2.1.12版本中测试可以正常写入
```
INSERT INTO twitter_test.`tweets` (`tweet_id`,`twitter_id`,`text`,`reference_tweet_type`,`reference_tweet_id`,`created_at`,`updated_at`,`retweet_count`,`reply_count`,`like_count`,`quote_count`) VALUES 
    -> ('1612944629728514048','64844802','RT @kimberlywtan: 📢✨New year, new team! @a16z is hiring (!!!) on our enterprise team. I cold-applied and you can too 🎉 Want to work with t…','retweeted','1612531671923826691','2023-01-10 22:49:01','2023-01-11 08:32:39.654',4,0,0,0)
    -> ,('1612928519297695751','64844802','RT @a16zBioHealth: In our latest Bio Eats World episode, @vijaypande and @pmarca dive into expert AI: As data and complexities for physicia…','retweeted','1612479444261773312','2023-01-10 21:45:00','2023-01-11 08:32:39.654',7,0,0,0)
    -> ,('1612905084912037888','64844802','RT @a16zcrypto: 22 of our most popular posts, podcasts, videos, and other resources from 2022. 🧵','retweeted','1612885694523019270','2023-01-10 20:11:53','2023-01-11 08:32:39.654',122,0,0,0)
    -> ,('1612882796493606921','64844802','RT @andrewchen: Excited to be working with Marcus Jacobs and Adam Schaub on their new startup :) Marcus was the GM of the Candy Crush fran…','retweeted','1612867346414317569','2023-01-10 18:43:19','2023-01-11 08:32:39.654',21,0,0,0)
    -> ,('1612865033381318693','64844802','RT @stevesi: #CES2023 — More Business, Less Spectacle. I walked 65,000 steps so you don\'t have to.It was a different year for CES. It…','retweeted','1612344855711338497','2023-01-10 17:32:44','2023-01-11 08:32:39.654',29,0,0,0)
    -> ,('1612834177186922499','64844802','RT @seidrofficial: We are thrilled to announce that we have closed a seed round led by @a16z. Together we are dedicated to crafting the nex…','retweeted','1612812209046040578','2023-01-10 15:30:07','2023-01-11 08:32:39.654',53,0,0,0)
    -> ,('1612814445222285314','64844802','RT @JoshLu: I\'m delighted to announce @a16z\'s seed investment in @seidrofficial, my first investment at the firm! Seidr is a new studio out…','retweeted','1612812258526265352','2023-01-10 14:11:43','2023-01-11 08:32:39.654',47,0,0,0)
    -> ,('1612812837163880449','64844802','Were excited to lead the seed investment in @seidrofficial. Founders Adam Schaub and Marcus Jacobs are building the next generation of platform-agnostic, cross-play games.Read more from a16z Games @andrewchen &amp; @joshlu ↓ https://t.co/0nGjaLyb7c','','','2023-01-10 14:05:19','2023-01-11 08:32:39.654',29,2,70,2);
ERROR 1064 (HY000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '<EOF>' at line 9
```

单独插入第九行 可以成功：
```
mysql> INSERT INTO twitter_connector_test.`tweets` (`tweet_id`,`twitter_id`,`text`,`reference_tweet_type`,`reference_tweet_id`,`created_at`,`updated_at`,`retweet_count`,`reply_count`,`like_count`,`quote_count`) VALUES 
    -> ('1612812837163880449','64844802','Were excited to lead the seed investment in @seidrofficial. Founders Adam Schaub and Marcus Jacobs are building the next generation of platform-agnostic, cross-play games.Read more from a16z Games @andrewchen &amp; @joshlu ↓ https://t.co/0nGjaLyb7c','','','2023-01-10 14:05:19','2023-01-11 08:32:39.654',29,2,70,2);
Query OK, 1 row affected (0.07 sec)
{'label':'insert_a63e269e-919c-11ed-a7d5-02369d384f1f', 'status':'VISIBLE', 'txnId':'232297810'}
```
单独插入前八行可以成功：
```
 INSERT INTO twitter_connector_test.`tweets` (`tweet_id`,`twitter_id`,`text`,`reference_tweet_type`,`reference_tweet_id`,`created_at`,`updated_at`,`retweet_count`,`reply_count`,`like_count`,`quote_count`) VALUES 
    -> ('1612944629728514048','64844802','RT @kimberlywtan: 📢✨New year, new team! @a16z is hiring (!!!) on our enterprise team. I cold-applied and you can too 🎉 Want to work with t…','retweeted','1612531671923826691','2023-01-10 22:49:01','2023-01-11 08:32:39.654',4,0,0,0)
    -> ,('1612928519297695751','64844802','RT @a16zBioHealth: In our latest Bio Eats World episode, @vijaypande and @pmarca dive into expert AI: As data and complexities for physicia…','retweeted','1612479444261773312','2023-01-10 21:45:00','2023-01-11 08:32:39.654',7,0,0,0)
    -> ,('1612905084912037888','64844802','RT @a16zcrypto: 22 of our most popular posts, podcasts, videos, and other resources from 2022. 🧵','retweeted','1612885694523019270','2023-01-10 20:11:53','2023-01-11 08:32:39.654',122,0,0,0)
    -> ,('1612882796493606921','64844802','RT @andrewchen: Excited to be working with Marcus Jacobs and Adam Schaub on their new startup :) Marcus was the GM of the Candy Crush fran…','retweeted','1612867346414317569','2023-01-10 18:43:19','2023-01-11 08:32:39.654',21,0,0,0)
    -> ,('1612865033381318693','64844802','RT @stevesi: #CES2023 — More Business, Less Spectacle. I walked 65,000 steps so you don\'t have to.It was a different year for CES. It…','retweeted','1612344855711338497','2023-01-10 17:32:44','2023-01-11 08:32:39.654',29,0,0,0)
    -> ,('1612834177186922499','64844802','RT @seidrofficial: We are thrilled to announce that we have closed a seed round led by @a16z. Together we are dedicated to crafting the nex…','retweeted','1612812209046040578','2023-01-10 15:30:07','2023-01-11 08:32:39.654',53,0,0,0)
    -> ,('1612814445222285314','64844802','RT @JoshLu: I\'m delighted to announce @a16z\'s seed investment in @seidrofficial, my first investment at the firm! Seidr is a new studio out…','retweeted','1612812258526265352','2023-01-10 14:11:43','2023-01-11 08:32:39.654',47,0,0,0);
Query OK, 7 rows affected (0.04 sec)
{'label':'insert_cb4e86af-919c-11ed-a7d5-02369d384f1f', 'status':'VISIBLE', 'txnId':'232298215'}
```

将第九行字符串中的; 去掉 也可以成功：
```
mysql> INSERT INTO twitter_connector_test.`tweets` (`tweet_id`,`twitter_id`,`text`,`reference_tweet_type`,`reference_tweet_id`,`created_at`,`updated_at`,`retweet_count`,`reply_count`,`like_count`,`quote_count`) VALUES 
    -> ('1612944629728514048','64844802','RT @kimberlywtan: 📢✨New year, new team! @a16z is hiring (!!!) on our enterprise team. I cold-applied and you can too 🎉 Want to work with t…','retweeted','1612531671923826691','2023-01-10 22:49:01','2023-01-11 08:32:39.654',4,0,0,0)
    -> ,('1612928519297695751','64844802','RT @a16zBioHealth: In our latest Bio Eats World episode, @vijaypande and @pmarca dive into expert AI: As data and complexities for physicia…','retweeted','1612479444261773312','2023-01-10 21:45:00','2023-01-11 08:32:39.654',7,0,0,0)
    -> ,('1612905084912037888','64844802','RT @a16zcrypto: 22 of our most popular posts, podcasts, videos, and other resources from 2022. 🧵','retweeted','1612885694523019270','2023-01-10 20:11:53','2023-01-11 08:32:39.654',122,0,0,0)
    -> ,('1612882796493606921','64844802','RT @andrewchen: Excited to be working with Marcus Jacobs and Adam Schaub on their new startup :) Marcus was the GM of the Candy Crush fran…','retweeted','1612867346414317569','2023-01-10 18:43:19','2023-01-11 08:32:39.654',21,0,0,0)
    -> ,('1612865033381318693','64844802','RT @stevesi: #CES2023 — More Business, Less Spectacle. I walked 65,000 steps so you don\'t have to.It was a different year for CES. It…','retweeted','1612344855711338497','2023-01-10 17:32:44','2023-01-11 08:32:39.654',29,0,0,0)
    -> ,('1612834177186922499','64844802','RT @seidrofficial: We are thrilled to announce that we have closed a seed round led by @a16z. Together we are dedicated to crafting the nex…','retweeted','1612812209046040578','2023-01-10 15:30:07','2023-01-11 08:32:39.654',53,0,0,0)
    -> ,('1612814445222285314','64844802','RT @JoshLu: I\'m delighted to announce @a16z\'s seed investment in @seidrofficial, my first investment at the firm! Seidr is a new studio out…','retweeted','1612812258526265352','2023-01-10 14:11:43','2023-01-11 08:32:39.654',47,0,0,0)
    -> ,('1612812837163880449','64844802','Were excited to lead the seed investment in @seidrofficial. Founders Adam Schaub and Marcus Jacobs are building the next generation of platform-agnostic, cross-play games.Read more from a16z Games @andrewchen &amp @joshlu ↓ https://t.co/0nGjaLyb7c','','','2023-01-10 14:05:19','2023-01-11 08:32:39.654',29,2,70,2);
Query OK, 8 rows affected (0.03 sec)
{'label':'insert_ef6458f8-919c-11ed-a7d5-02369d384f1f', 'status':'VISIBLE', 'txnId':'232298569'}
```

**Comment by rickif** [comment]
@duopa Could you please provide your statement to create the table?

**Comment by zhuangzhuangdo** [comment]
```
CREATE TABLE `tweets` (
  `tweet_id` varchar(50) NOT NULL COMMENT "",
  `twitter_id` varchar(20) NOT NULL COMMENT "",
  `text` varchar(65533) NULL COMMENT "",
  `reference_tweet_type` varchar(50) NULL COMMENT "",
  `reference_tweet_id` varchar(50) NULL COMMENT "",
  `created_at` datetime NOT NULL COMMENT "",
  `updated_at` datetime NOT NULL COMMENT "",
  `retweet_count` bigint(20) NOT NULL COMMENT "",
  `reply_count` bigint(20) NOT NULL COMMENT "",
  `like_count` bigint(20) NOT NULL COMMENT "",
  `quote_count` bigint(20) NOT NULL COMMENT ""
) ENGINE=OLAP 
UNIQUE KEY(`tweet_id`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`tweet_id`) BUCKETS 40 
PROPERTIES (
"replication_num" = "3",
"in_memory" = "false",
"storage_format" = "V2",
"enable_persistent_index" = "false"
);
```

**Comment by duopa** [comment]
```
CREATE TABLE `tweets` (
  `tweet_id` varchar(50) NOT NULL COMMENT "",
  `twitter_id` varchar(20) NOT NULL COMMENT "",
  `text` varchar(65533) NULL COMMENT "",
  `reference_tweet_type` varchar(50) NULL COMMENT "",
  `reference_tweet_id` varchar(50) NULL COMMENT "",
  `created_at` datetime NOT NULL COMMENT "",
  `updated_at` datetime NOT NULL COMMENT "",
  `retweet_count` bigint(20) NOT NULL COMMENT "",
  `reply_count` bigint(20) NOT NULL COMMENT "",
  `like_count` bigint(20) NOT NULL COMMENT "",
  `quote_count` bigint(20) NOT NULL COMMENT ""
) ENGINE=OLAP 
UNIQUE KEY(`tweet_id`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`tweet_id`) BUCKETS 40 
PROPERTIES (
"replication_num" = "3",
"in_memory" = "false",
"storage_format" = "V2",
"enable_persistent_index" = "false"
);
```

**Comment by yangrong688** [comment]
this issue seems to have the same problem with https://github.com/StarRocks/starrocks/issues/16552
@rickif please also have a look at this issue, thanks.

**Comment by rickif** [comment]
This problem is fixed in 2.5. We are trying to port the fix to 2.3.

**Comment by duopa** [comment]
There are also problems with this version : 2.2.10

**Comment by yangrong688** [comment]
> 
Hi @rickif , is there any progress for fix this issue?

**Comment by rickif** [comment]
@HangyuanLiu PTAL.

**Comment by yangrong688** [comment]
Hi @rickif, @HangyuanLiu, @duopa. I've found this issue have been fixed since #11752, but it's hard to backport to older branch because legcy paser still in use in branch 2.2 -> 2.4.
A minor change in #17846 could fix this issue, PTAL.
